blendo/sourcesSync/sources-sync.go

```diff
@@ -0,0 +1,245 @@
+						destination.Enabled) {
+						destinationNewCount++
+						// if destination or source enabled or connection status or configuration has change then update registry configuration
+					}
+				}
+			}
+			// if destination not found in the previous destinations, means is new
+			if !destinationExists && !destination.Deleted && destination.IsConnectionEnabled {
+				destinationNewCount++
+			}
+		}
+	}
+	sourceDestinationsLength := len(source.Destinations)
+	prevDestinationLength := len(prevSource.Destinations)
+	if destinationDeletedCount == sourceDestinationsLength && destinationNewCount == 0 {
+		br.SourcesClient.DeleteConfigFromRegistry(pipelineID)
+		return true
+	} else if prevDestinationLength == prevDestinationDeletedCount && destinationNewCount > 0 {
+		br.SourcesClient.PutConfigToRegistry(pipelineID, br.getConfig(source))
+		return true
+	}
+	return false
+}
+
+// calculateDifferencesAndUpdateRegistry calculate the differences from the previous config and make the coresponding actions
+// we need to update blendo registry when the destination or source configuration or enabled status change
+// and delete from blendo registry when the connection is no longer exists or a source or destination is deleted
+func (br *SourcesSync) calculateDifferencesAndUpdateRegistry(sources []backendconfig.SourceT) {
+	prevSources := br.currentSourceJSON.Sources
+	for _, source := range sources {
+		pipelineID := br.getPipelineID(source.ID)
+		if source.SourceDefinition.Category != "cloud" {
+			continue
+		}
+		sourceExists := false
+		if br.isConfigSet {
+			for _, prevSource := range prevSources {
+				if prevSource.ID == source.ID {
+					sourceExists = true
+					if source.Deleted != prevSource.Deleted || (source.Enabled != prevSource.Enabled && !source.Enabled) {
+						br.SourcesClient.DeleteConfigFromRegistry(pipelineID)
+						continue
+					} else if !br.checkSourceDestinations(source, prevSource) {
+						if !reflect.DeepEqual(source.Config, prevSource.Config) || (source.Enabled != prevSource.Enabled && source.Enabled) {
+							br.SourcesClient.PutConfigToRegistry(pipelineID, br.getConfig(source))
+						}
+					}
+				}
+			}
+		}
+		// if source not found in the previous sources mean is new
+		if !sourceExists {
+			destinationDeletedCount := 0
+			for _, destination := range source.Destinations {
+				if !destination.IsConnectionEnabled || destination.Deleted {
+					destinationDeletedCount++
+					// if destination or source enabled or connection status or configuration has change then update registry configuration
+				}
+			}
+			if destinationDeletedCount == len(source.Destinations) {
+				br.SourcesClient.DeleteConfigFromRegistry(pipelineID)
+			} else {
+				br.SourcesClient.PutConfigToRegistry(pipelineID, br.getConfig(source))
+			}
+		}
+	}
+}
+
+// getResources returns the resources of a pipeline in form that registry understands
+func (br *SourcesSync) getResources(resources []interface{}) map[string]ResourceT {
```

Comment:
This should not be part of *SourcesSync, as it is not using it.
The comment is unclear, what is a pipeline in the context of this function?